server/src/database.rs

```diff
@@ -0,0 +1,162 @@
+use actix::prelude::*;
+use futures::stream::Stream;
+use l337_postgres::PostgresConnectionManager;
+use std::convert::TryFrom;
+use std::fs;
+use tokio_postgres::row::Row;
+use tokio_postgres::NoTls;
+use vertex_common::UserId;
+
+pub struct GetUserById(pub UserId);
+
+impl Message for GetUserById {
+    type Result = Result<Option<User>, l337::Error<tokio_postgres::Error>>;
+}
+
+pub struct GetUserByName(pub String);
+
+impl Message for GetUserByName {
+    type Result = Result<Option<User>, l337::Error<tokio_postgres::Error>>;
+}
+
+pub struct CreateUser(pub User);
+
+impl Message for CreateUser {
+    type Result = Result<(), l337::Error<tokio_postgres::Error>>;
+}
+
+pub struct User {
+    pub id: UserId,
+    pub name: String,
+    pub password_hash: String,
+}
+
+impl TryFrom<Row> for User {
+    type Error = tokio_postgres::Error;
+
+    fn try_from(row: Row) -> Result<User, tokio_postgres::Error> {
+        Ok(User {
+            id: UserId(row.try_get("id")?),
+            name: row.try_get("name")?,
+            password_hash: row.try_get("password_hash")?,
+        })
+    }
+}
+
+pub struct DatabaseServer {
+    pool: l337::Pool<PostgresConnectionManager<NoTls>>,
+}
+
+impl DatabaseServer {
+    pub fn new(sys: &mut SystemRunner) -> Self {
+        let mgr = PostgresConnectionManager::new(
+            fs::read_to_string("db.conf")
+                .expect("db.conf not found")
+                .parse()
+                .unwrap(),
+            NoTls,
+        );
+
+        let pool = sys
+            .block_on(l337::Pool::new(mgr, Default::default()))
+            .expect("db error");
+        DatabaseServer { pool }
+    }
+}
+
+impl DatabaseServer {
+    fn create_tables(&mut self) -> impl Future<Item = (), Error = ()> {
+        self.pool
+            .connection()
+            .and_then(|mut conn| {
+                conn.client
+                    .prepare(
+                        "CREATE TABLE IF NOT EXISTS users (
```

Comment:
Maybe should make these their own files and `include_str!`

Reply:
Maybe indeed

Reply:
May also be clearer inline